### ISSUE TITLE
Weird thing happen when using torch[torch] after tensor2trt

### ISSUE THREAD

**Issue by xvolica** [NVIDIA-AI-IOT/torch2trt#671]
I find that using torch as index to select some lines from tensor will cause dimension change, even data loss.
For example here is my code:
```
import torch
import torch.nn as nn
from torch2trt import torch2trt 

print(torch.__version__)
a = torch.eye(6)
print("======befroe======")
k = torch.ones((1,6)).view(-1)>0.5
print("shape a[k]", a[k].size())
i = torch.ones((1), dtype=torch.long)
print("shape a[1]", a[i].size())
ii = torch.ones((2), dtype=torch.long)
print("shape a[1,1]", a[ii].size())

print("======just do convert======")
net = nn.Sequential(
    nn.Conv2d(3, 5, 3),
    nn.Conv2d(5, 5, 3)
)
x = torch.ones((1, 3, 224, 224))
model_trt = torch2trt(net.cuda(), [x.cuda()])

print("======after======")
print("shape a[k]", a[k].size())
print("shape a[1]", a[i].size())
print("shape a[1,1]", a[ii].size())
```
And the output:
```
1.8.0+cu111
======befroe======
shape a[k] torch.Size([6, 6])
shape a[1] torch.Size([1, 6])
shape a[1,1] torch.Size([2, 6])
======just do convert======
======after======
shape a[k] torch.Size([1, 6, 6])
shape a[1] torch.Size([6])
shape a[1,1] torch.Size([])
```

But when I change all the index to list type, the results are same as before part, so what's the reason?
